engine/src/main/java/org/hibernate/search/spi/impl/ConcurrentIndexedTypeMap.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.spi.impl;
+
+import java.util.Collection;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.hibernate.search.spi.IndexedTypeIdentifier;
+import org.hibernate.search.spi.IndexedTypeMap;
+import org.hibernate.search.spi.IndexedTypesSet;
+
+public class ConcurrentIndexedTypeMap<V> implements IndexedTypeMap<V> {
```

Comment:
This whole thing is temporary, I already have several other candidate prototypes for later..

It's a placeholder and I'd prefer to just avoid conflicts for now.

Once it's in, you'll also be able to propose changes as you prefer.